lib/pure/os.nim

```diff
@@ -2266,16 +2266,12 @@ iterator walkDir*(dir: string; relative = false, checkDir = false):
   ##        / fileA2.txt
   ##
   ## and this code:
-  ##
-  ## .. code-block:: Nim
-  ##     for kind, path in walkDir("dirA"):
-  ##       echo(path)
-  ##
-  ## produce this output (but not necessarily in this order!)::
-  ##   dirA/dirB
-  ##   dirA/dirC
-  ##   dirA/fileA1.txt
-  ##   dirA/fileA2.txt
+  runnableExamples("-r:off"):
```

Comment:
replace
```nim
  else:
    when weirdTarget:
      for k, v in items(staticWalkDir(dir, relative)):
        yield (k, v)
    elif defined(windows):
```
by:
```nim
  else:
    when defined(nimscript): discard # taken care of by nimvm but this branch is needed
    elif defined(js):
      doAssert false, "not implemented" # xxx support nodejs in future
    elif defined(windows):
```
so that we address the bug mentioned in https://github.com/nim-lang/Nim/issues/16553#issue-777473092

> this callback gets called when weirdTarget aka nimscript or JS is true, meaning this will also be used in JS

(i tried, this makes things work with 
nim r main
nim e main
nim r -b:js main)

Reply:
follow up PRs instead

Reply:
ok